src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2187,23 +2243,25 @@ class CgroupUtils(object):
         if not pids:
             return
         # check pids to make sure that they are owned by the job owner
-        if PBS_VERSION[0] >= 13:
-            if pbs.event().type == pbs.EXECJOB_ATTACH:
-                pbs.logmsg(pbs.EVENT_DEBUG4, "event type: attach")
-                try:
-                    uid = pwd.getpwnam(pbs.event().job.euser).pw_uid
-                except:
+        if pbs.event().type == pbs.EXECJOB_ATTACH:
+            pbs.logmsg(pbs.EVENT_DEBUG4, "event type: attach")
+            try:
+                uid = pwd.getpwnam(pbs.event().job.euser).pw_uid
+            except KeyboardInterrupt:
+                raise
+            except:
+                pbs.logmsg(pbs.EVENT_DEBUG2,
+                           "Failed to lookup UID by name")
+                raise
```

Comment:
Since you raise the exception here, you don't necessarily need the KeyboardInterrupt exception by itself.  You would get this log message.  Up to you.

Reply:
We don't really want the log message if a KeyboardInterrupt is received because it wasn't a failure in looking up the UID, but rather an asynchronous event. I'd prefer to leave this as is.